src/de.rs

```diff
@@ -4,14 +4,20 @@ use std::fmt;
 use std::str::Utf8Error;
 use std::string::FromUtf8Error;
 
-use parse::Bytes;
+use parse::{Bytes, Position};
 
 use serde::de::{self, Deserializer as Deserializer_, DeserializeSeed, Visitor};
 
 pub type Result<T> = ::std::result::Result<T, Error>;
 
 #[derive(Clone, Debug, PartialEq)]
 pub enum Error {
+    Message(String),
+    Parser { kind: ErrorKind, pos: Position },
```

Comment:
nit: doesn't look like names are useful here, could just be `Parser(ErrorKind, Position)`